### ISSUE TITLE
data-mapping 5.0.4 multi column unique constraint causes an error: UniqueKey already exists!

### ISSUE THREAD

**Issue by salex772** [grails/grails-data-mapping#703]
Consider this domain class

```
class UserPaySystem implements Serializable
{
	static belongsTo = [user: User]
	static hasOne    = [paySystem: PaySystem]

	static constraints = {
		paySystem nullable: false
		account nullable: false, blank: false, unique: ['user', 'paySystem']
	}

	static mapping = {
		table '`tbl_user_payment_system`'
		paySystem fetch: 'join'
	}

	String account

	Date dateCreated
	Date lastUpdated
}
```

This produces this error:
   
    Error creating bean with name 'sessionFactory': Invocation of init method failed; nested exception is org.hibernate.MappingException: UniqueKey UKfa8b86a17b3a14d1b68013695ea7 already exists!

Without 

     unique: ['user', 'paySystem']

everything work as expected

[grails315.zip](https://github.com/grails/grails-data-mapping/files/220852/grails315.zip)

**Comment by graemerocher** [comment]
This duplicates https://github.com/grails/grails-data-mapping/issues/703 I assume?

**Comment by graemerocher** [comment]
Sorry ignore me

**Comment by graemerocher** [comment]
Fixed by  #703
